### ISSUE TITLE
question: how `partial actor pattern` compares to X

### ISSUE THREAD

**Issue by dzmitry-lahoda** [paritytech/orchestra#86]
- https://www.firezone.dev/blog/sans-io
- https://en.wikipedia.org/wiki/Reactor_pattern
- https://docs.dapr.io/developing-applications/building-blocks/actors/actors-overview/
- Erlang OTP actors
- direct usage of tokio futures/streams/channels

**Comment by drahnr** [comment]
So `orchestra` is technically a set of microservices (=subsystems) within a binary, where each subsystem is responsible for scaling itself adequately to the sub-domain it covers, in practice this means using `FuturesUnordered` or dedicated threads. Note that orchestra has the concept of a `signal` that is essentially a global clock, defining relevance of actual work.

`orchestra` uses _a lot_ of channels under the hood to facilitate communication between subsystems, so there is buffering invovled which disambiguates it from message passing and classical actor patterns.
The reactor has a single driver, `orchestra` each subsystem consumes messages, so technically a multi-reactor.

tl;dr orchestra is a mix of all of the referenced, it's _very_ domain specific and I have strong doubts it being useful outside of it's original domain.
